index.js

```diff
@@ -0,0 +1,49 @@
+// constants
+const EN_LIST = ['en-US', 'en-GB', 'en', 'en-AU'];
+module.exports.ARGUMENTS = ['--restrict-filenames', '--write-sub', '--write-auto-sub', '--max-downloads', '1', '-f', '[height <=? 720][format_id != source]', '-j'];
+
+class YtdlProcess {
+  constructor(origin) {
+    this.origin = origin;
+  }
+
+  process(output) {
+    let data;
+    try {
+      data = JSON.parse(output.toString().trim());
+    } catch (err) {
+      return null;
+    }
+
+    const subtitleFile = findBestSubtitleFile(data.subtitles) || findBestSubtitleFile(data.automatic_captions);
+
+    let subtitleUrl;
+    if (subtitleFile) {
+      subtitleUrl = `${this.origin}/ytdl/vtt?suburi=${encodeURIComponent(subtitleFile.subs.url)}`;
+    }
+
+    return {
+      title: data.title,
+      url: data.url,
+      duration: data.duration,
+      subtitle_url: subtitleUrl,
+      cookies: data.http_headers.Cookie
+    };
+  }
+}
+
+module.exports = YtdlProcess;
```

Comment:
doesn't this mean ARGUMENTS is not longer available on module.exports?

Reply:
Nope, it's confusing but the main `require()` will be the class and require().ARGUMENTS` will be the `ARGUMENTS`

Reply:
```
jake@dubnium 0+ ~/vivi/ytdl-process vivi-1342-processing-module
 $ ls
index.js  package.json  README.md  yarn.lock

jake@dubnium 0+ ~/vivi/ytdl-process vivi-1342-processing-module
 $ node
> const ytdlProcess = require('./index');
undefined
> ytdlProcess.ARGUMENTS
undefined
```

Reply:
:facepalm: I guess I'm crazy, you have to have a `static get()` on the class

Reply:
yeah, or just don't use a class :smile:

Reply:
OR you can export the class as a property instead of the whole export